src/osd/PG.cc

```diff
@@ -4011,6 +4027,16 @@ void PG::chunky_scrub(ThreadPool::TPHandle &handle)
 	  scrubber.reserved_peers.clear();
 	}
 
+	{
+	  ObjectStore::Transaction t;
+	  if (scrubber.store) {
+	    scrubber.store->cleanup(&t);
+	  }
```

Comment:
https://github.com/athanatos/ceph/tree/wip-13505-sam has a version of something that I think would work.